xtests/tst-pam_motd3.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+TST_DIR="tst-pam_motd3.d"
+
+function tst_cleanup() {
+    rm -rf "${TST_DIR}"
+    rm -f tst-pam_motd3.out
+}
+
+mkdir -p ${TST_DIR}
+mkdir -p ${TST_DIR}/etc/motd.d
+mkdir -p ${TST_DIR}/run/motd.d
+mkdir -p ${TST_DIR}/usr/lib/motd.d
+
+# Verify motd is still displayed when not overridden
+echo "motd: test-show in run - show" > ${TST_DIR}/run/motd.d/test-show.motd
+
+# Test hidden by a symlink to another file (which should not be shown)
+echo "motd: hidden-by-symlink in usr/lib - not show" > ${TST_DIR}/usr/lib/motd.d/hidden-by-symlink.motd
+echo "motd: test-from-symlink - not show" > ${TST_DIR}/test-from-symlink.motd
+ln -s ${TST_DIR}/test-from-symlink.motd ${TST_DIR}/run/motd.d/hidden-by-symlink.motd
+
+# Test hidden by a null symlink
```

Comment:
Right, previously this was done implicitly, but by considering *any* symlink to override a file and following the symlink. It probably is better to check the symlink target for `/dev/null` and not try to print it - working this in

Reply:
Actually - leaving this done implicitly - think it is fine. This is to avoid an extra check like `readlink(2)` being necessary before displaying the motd files

Reply:
Having it done implicitly is OK.